### PR TITLE
feat: Adapt time to fiori3

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/577

## Description
There is whole time component refactored, right now it follows all of fiori3 specs. Also there are SAP theming parameters used.

## Breaking Changes: 
Refactored markup for time/time picker.
Removed Classes: 
* `fd-time__control`

New modes:
* `fd-time(--compact/--tablet/--collapsible)`

New elements:
* `fd-time__col`
* `fd-time__slider-label`
* `fd-time__wrapper(--active/--meridian)`
* `fd-time__item(--collapsed)`
* `fd-time__unit`

### Before:
image

TimePicker:
image
image

### After:
Cozy:
image

Tablet:
image

Compact:
image

Time Picker:
image
image
image

### CODE REVIEW

File: src/time.scss [hunk 1]
```diff
@@ -7,31 +7,221 @@ $block: #{$fd-namespace}-time;
 .#{$block} {
   @include fd-reset();
 
-  width: 184px; // Quick fix to solve overflow issue
+  $fd-time-item-height: 2.875rem;
```
Comment (stefanoScalzo) [inline src/time.scss]: should be 2 rem for cozy mode, it only gets bigger for smaller applications like tablet and phone

File: src/time.scss [hunk 2]
```diff
@@ -7,31 +7,221 @@ $block: #{$fd-namespace}-time;
...
+  }
+
+  &__slider-label {
+    @include fd-reset();
+
+    line-height: $fd-time-item-compact-height;
+    height: $fd-time-item-compact-height;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
```
Comment (stefanoScalzo) [inline src/time.scss]: we need to define a max-width for the overflow, ask the designers

**Comment by droshev** [comment]
@trilodge can you review the PR from a11y point of view?

**Comment by droshev** [comment]
we can review but merge once version `0.8.0` is release

File: src/time.scss (hunk 1 above)
Comment (JKMarkowski) [reply src/time.scss:10]: Hello @stefan-sto compact mode has 2rem, default one has 2.875rem

File: src/time.scss (hunk 2 above)
Comment (JKMarkowski) [reply src/time.scss:158]: Hello @stefan-sto I think max width should be `100%` there

File: src/time.scss [hunk 3]
```diff
@@ -7,31 +7,234 @@ $block: #{$fd-namespace}-time;
...
+  &__col {
     @include fd-reset();
 
-    display: inline-block;
-    max-width: fd-space(10);
-    text-align: center;
-    margin-right: fd-space(1);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
```
Comment (InnaAtanasova) [inline src/time.scss:26]: You can use fd-flex-vertical-center() {flex-direction: column;}

File: src/time.scss [hunk 4]
```diff
@@ -7,31 +7,234 @@ $block: #{$fd-namespace}-time;
...
+
+  &__wrapper {
+    @include fd-reset();
+
+    overflow: hidden;
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
```
Comment (InnaAtanasova) [inline src/time.scss:56]: You can use @include fd-flex-center() {flex-direction: column;}

File: src/time.scss [hunk 5]
```diff
@@ -7,31 +7,234 @@ $block: #{$fd-namespace}-time;
...
+  &__item {
+    @include fd-reset();
+
+    cursor: default;
+    width: 3rem;
+    height: $fd-time-item-height;
+    min-height: $fd-time-item-height;
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (InnaAtanasova) [inline src/time.scss:92]: @include fd-flex-center()

File: src/time.scss [hunk 6]
```diff
@@ -7,31 +7,234 @@ $block: #{$fd-namespace}-time;
...
+  &__unit {
+    @include fd-reset();
+
+    text-align: center;
+    color: inherit;
+    display: block;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    align-items: center;
+    justify-content: center;
```
Comment (InnaAtanasova) [inline src/time.scss:146]: Is this working with `display: block`?

File: src/time.scss [hunk 7]
```diff
@@ -7,31 +7,234 @@ $block: #{$fd-namespace}-time;
...
+  }
+
+  &__slider-label {
+    @include fd-reset();
+
+    line-height: $fd-time-item-compact-height;
+    height: $fd-time-item-compact-height;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
```
Comment (InnaAtanasova) [inline src/time.scss:167]: You can use @include fd-ellipsis()

File: src/time.scss (hunk 1 above)
Comment (InnaAtanasova) [reply src/time.scss:10]: @JKMarkowski I think this is the wrong Stefano ;)

File: src/time.scss (hunk 2 above)
Comment (InnaAtanasova) [reply src/time.scss:167]: @JKMarkowski wrong Stefano

**Comment by InnaAtanasova** [comment]
I don't think it's allowed to have this situation

Screen Shot 2020-04-16 at 4 34 13 PM

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-04-16 at 4 37 50 PM

The width of the label should be the same as the sliders

**Comment by JKMarkowski** [comment]
> I don't think it's allowed to have this situation
> 
> Screen Shot 2020-04-16 at 4 34 13 PM

There is no way to handle issues like that in pure css

File: src/time.scss (hunk 6 above)
Comment (JKMarkowski) [reply src/time.scss:146]: Leftovers :D

**Comment by InnaAtanasova** [comment]
> > I don't think it's allowed to have this situation
> > Screen Shot 2020-04-16 at 4 34 13 PM
> 
> There is no way to handle issues like that in pure css

I was afraid you will say so :D We will fix it in NGX then

**Comment by InnaAtanasova** [comment]
Looks great, just 1 tiny issue with RTL (the text is not centered)

Screen Shot 2020-04-20 at 5 01 59 PM

**Comment by JKMarkowski** [comment]
@InnaAtanasova I just removed `fd-form-label` class usage, and put mixin instead. 
@rengare Variables added

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-04-23 at 11 54 13 AM
Screen Shot 2020-04-23 at 11 55 01 AM
Screen Shot 2020-04-23 at 11 56 21 AM
check the playground to normalized and check the theming.

**Comment by JKMarkowski** [comment]
@stefanoScalzo  Everything works fine, it's just about the uglify variables for tests
image

File: src/time.scss [hunk 8]
```diff
@@ -7,31 +7,221 @@ $block: #{$fd-namespace}-time;
...
     @include fd-reset();
 
-    display: inline-block;
-    max-width: fd-space(10);
-    text-align: center;
-    margin-right: fd-space(1);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    margin-right: 0.5rem;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply src/time.scss]: Done

**Pull request merged**